### PR TITLE
Enable source map generation in ts-loader compilerOptions

### LINKED ISSUE #4382: Source maps not working properly
Something has regressed about our source maps. In a dev build, sources in the chrome dev tools look like this:

image

On studio.foxglove.dev they look like this:

image

Neither of these actually looks like the original source file.

### PR DESCRIPTION
**Pull request by jtbandes**
**User-Facing Changes**
None

**Description**
Our packages/studio-base/tsconfig.json has source maps disabled. This is probably because we don't want to bother outputting source maps in the published studio-base package. However, we do want them enabled for Studio builds.

Fixes #4382

### CODE REVIEW

File: desktop/integration-test/buildSize.test.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import * as path from "path";
 
 // Adjust this byte size as needed if the app is growing for a valid reason.
 // This check is here to catch unexpected ballooning of the build size
-const MAX_BUILD_SIZE = 150_000_000;
+const MAX_BUILD_SIZE = 160_000_000;
```
Comment (defunctzombie) [inline desktop/integration-test/buildSize.test.ts:10]: :(

File: package.json [hunk 2]
```diff
@@ -57,7 +57,6 @@
   },
   "resolutions": {
     "@types/node": "15.3.0",
-    "esbuild-loader/esbuild": "0.14.7",
```
Comment (defunctzombie) [inline package.json:60]: Do we no longer want to force a specific esbuild anymore?

**Review by defunctzombie — APPROVED** [review]

File: desktop/integration-test/buildSize.test.ts (hunk 1 above)
Comment (jtbandes) [reply desktop/integration-test/buildSize.test.ts:10]: :)

File: package.json (hunk 2 above)
Comment (jtbandes) [reply package.json:60]: Unclear why this was here? Things seem to work when I remove it and upgrade packages.

Comment (jtbandes) [reply package.json:60]: https://github.com/foxglove/studio/pull/2480

**Pull request merged**